browser/js/identities.js

```diff
@@ -30,42 +30,48 @@ const identityMatchers = [
   {
     service: "keybase",
     getUsername: function(loc) { return loc.pathname.split('/')[1]; },
-    locationMatches: new RegExp('\\.keybase\\.(?:io|pub)/([\\w]+)[/]?'),
-    originAndPathMatches: '\\.keybase\\.(io|pub)/[\\w]+[/]?',
+    locationMatches: new RegExp('\\.keybase\\.io/([\\w]+)[/]?'),
+    originAndPathMatches: 'keybase\\.io/[\\w]+[/]?',
```

Comment:
We still have the code that prefixes a period for naked domains in matchService, any reason for getting rid of the `\\.` prefix on some of the matchers? (Reddit still has it for some reason, the rest go stripped.)

Also could add `^\\.` for bonus safety.

Reply:
The problem is the declarative matchers -- for keybase, github & twitter the regex won't match. for reddit and facebook i left it in since they redirect to `www`.  we can add `^` and `^www` to the `locationMatches` but the `originAndPathMatches` seems to reject it :/

Reply:
Hey @joshblum and @shazow,
 
After the #10277 PR was merged, I filed an issue at [https://hackerone.com/keybase](https://hackerone.com/keybase) as the patch was indeed insufficient to validate the hostname (a security vulnerability was still present). 

I even attached a patch in the report, but I didn't hear back from @maxtaco. So you might want to take a look at the proposed patch in the HackerOne [report](https://hackerone.com/reports/309567) too.